ethrpc/ethrpc.go

```diff
@@ -58,17 +59,17 @@ func (s *Provider) SetHTTPClient(httpClient *http.Client) {
 	s.httpClient = httpClient
 }
 
-func (p *Provider) Do(ctx context.Context, calls ...Call) error {
+func (p *Provider) Do(ctx context.Context, calls ...Call) ([]byte, error) {
```

Comment:
Just an idea but wouldn't it be simpler and more ergonomic to, instead of returning the entire body bytes from `Do`, have new `CallBuilder`s whose `Into` accepts a `json.RawMessage`?

So, e.g. we would then have a function variant of:

```
func RawBlockByHash(hash common.Hash) CallBuilder[json.RawMessage]
```

This would IMO fit more into the existing design, and this way you can even batch requests with raw responses... like:

```
var m1, m2 json.RawMessage
err = p.Do(ctx, RawBlockByHash(h1).Into(&m1), RawBlockByHash(h2).Into(&h2))
```

What do you think?